src/android/HockeyApp.java

```diff
@@ -27,14 +31,69 @@
 import java.lang.Thread;
 import java.text.SimpleDateFormat;
 import java.util.Calendar;
+import java.util.ArrayList;
+
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.view.View;
 
 public class HockeyApp extends CordovaPlugin {
 
+    public static final long XWALK_SCREENSHOT_WAIT_MS = 5000;
+    public static final String XWALK_SCREENSHOT_CAPTURE_MSG = "captureXWalkBitmap";
+    public static final String XWALK_SCREENSHOT_BITMAP_MSG = "onGotXWalkBitmap";
+    
     public static boolean initialized = false;
     public static String appId;
+    public static Object monitor = new Object();
+    public static volatile Bitmap bitmap;
     
     private ConfiguredCrashManagerListener crashListener;
+    
+    // Integration with Crosswalk requires:
+    // - Crosswalk engine version 18 or later
+    // - Latest cordova-plugin-crosswalk-webview plugin     
+    private Bitmap getBitmap() {
+        bitmap = null;
+        boolean isCrosswalk = false;
+        try {
+            Class.forName("org.crosswalk.engine.XWalkWebViewEngine");
+            isCrosswalk = true;
+        } catch (Exception e) {
+        }
 
+        if (isCrosswalk) {
+            long start = System.currentTimeMillis();
+            webView.getPluginManager().postMessage(XWALK_SCREENSHOT_CAPTURE_MSG, this);
+            try {
+                synchronized (monitor) {
+                    monitor.wait(XWALK_SCREENSHOT_WAIT_MS);
+                }
+            } catch (InterruptedException ex) {
+                Thread.currentThread().interrupt();
```

Comment:
Actually, can we ex.printStackTrace() before re-interrupting. Just so we have better debug information if it ever happens.